lib/engine/game/g_18_mag.rb

```diff
@@ -427,6 +454,18 @@ def routes_subsidy(routes)
         routes.sum(&:subsidy)
       end
 
+      def reset_route!(route)
+        @round.snw_train = nil if route.train == @round.snw_train
```

Comment:
why dous the route have any affect on the round like this?

Reply:
I use round_state to communicate between the SpecialBuy actions that are used to purchase "powers" that affect the route (bonus revenues, plus-train upgrades, ability use mines) and the revenue calculations that are done in Route via game-specific methods.

Reply:
I needed to add a reset_route! method to force the powers to be recalculated when the user clears or resets the routes via the UI.

Reply:
Deleted. The "railcar" calculations are now stateless.